contactaccessrights/CRM/Contactaccessrights/BAO/Rights.php

```diff
@@ -30,57 +24,4 @@ public static function create($params) {
 
     return $instance;
   }
```

Comment:
Another reason is because it being a service allows it to be injected in the ContactAcessRightsMenu Group and makes it easy to be mocked in tests tests. Also I believe the method is more suited for a Service than a BAO because it involves getting information that is not related to the Rights DAO alone but joining to some other tables.

Reply:
It only joins with option value/groups tables so that it can filter by region/location. We do the same thing in L&A to filter Requests by status and all that is done in the BAO, not in the Service. In the end, it's just a method to return a list of Rights given certain criteria. The BAO should be responsible to know how to do that (think of BAOs as gateways to the database). Unfortunately, whoever implemented this made a few mistakes like return an array instead of objects and also including the option value information in the response. Ideally, this is what should have been fixed, but you don't need to worry about that for now.

About the tests, you can inject the BAO the same way as you've injected the service. This is just an instance method